pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -282,12 +282,23 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 
 		if isExposed(infinispan) {
 			externalService := r.serviceExternal(infinispan)
-			err = r.client.Create(context.TODO(), externalService)
-			if err != nil && !errors.IsAlreadyExists(err) {
-				reqLogger.Error(err, "failed to create external Service", "Service", ser)
-				return reconcile.Result{}, err
+			if externalService != nil {
+				err = r.client.Create(context.TODO(), externalService)
+				if err != nil && !errors.IsAlreadyExists(err) {
+					reqLogger.Error(err, "failed to create external Service", "Service", ser)
+					return reconcile.Result{}, err
+				}
+				infinispan.Spec.Expose = externalService.Spec
+				err = r.client.Update(context.TODO(), infinispan)
+				if err != nil {
+					reqLogger.Error(err, "failed to Infinispan with Service spec", "Service", ser)
+					return reconcile.Result{}, err
+				}
+				reqLogger.Info("Created External Service", "Service", externalService)
 			}
-			reqLogger.Info("Created External Service", "Service", externalService)
+		} else {
+			reqLogger.Info("External Service NOT created. Unsupported type?", "Service", nil)
```

Comment:
This part triggers when `isExposed == false` and `isExposed` checks `infinispan.Spec.Expose.Type != ""`.

This means that if user doesn't want service to be exposed (he doesn't define the expose type) then the Operator log will inform him that it didn't really created a service and it's maybe because it's type is unsupported. I think this may cause a confusion.

Reply:
yeah, the `else` should be in the inner block. Thanks!